### DISCUSSION TITLE
LibreWolf + KeepassXC-Browser + KeepassXC - connection failed on debian

### DISCUSSION THREAD

**Discussion by frank-github** [keepassxreboot/keepassxc#7669] [Q&A] [answered]
Hi,

try to connect LibreWolf-Browser to KeepassXC-DB with browser addon KeepassXC-Browser
but connections fails.
how can I collect hints on which end why the connection is not possible?

running debian bullseye (with backports)
keepassXC 2.6.6 as appimage 
keepassXC-Browser 1.7.11
LibreWolf v98.0.1-1

thanks for any hints and tips

**Comment by droidmonkey** [comment, marked as answer]
https://github.com/keepassxreboot/keepassxc-browser/wiki/Troubleshooting-guide

**Comment by lubboo** [comment]
I had the same problem with Librewolf and keepassxc (installed via synaptic) in Linux Mint 20.3.

The solution:

Searched for "native" with Nautilus in the user folder /home/user (show hidden files).
Several "native-messaging-hosts" folders were found in the browser folders, just not in /home/user/.librewolf.

I then copied the folder from .mozilla to .librewolf. The file "org.keepassxc.keepassxc_browser.json" must be included.

After that the Keepass browser addon got a connection to the database. I still had to assign a unique identifier (librewolf) for the connection. Then it worked.

Translated with www.DeepL.com/Translator (free version)

**Reply by infoener** [reply]
thanks too, work for me in opensuse tumbledweed

**Reply by zfkdnop** [reply]
Thanks. Worked for me on Linux Mint.
Copied ~/.mozilla/native-messaging-hosts into ~/.librewolf/. Restarted librewolf, connected KeepassXC-browser to my database (by clicking the Reload / Connect button, KeePassXC then prompted me to provide an identifier for the new connection and it worked flawlessly!

**Comment by keizah7** [comment]
solved on macos: https://github.com/keepassxreboot/keepassxc-browser/issues/1353#issuecomment-912416304

**Reply by nafarinha** [reply]
Regarding the solution below for macOS:
`ln -s ~/.mozilla/native-messaging-hosts ~/.librewolf/native-messaging-hosts`

It's not working on MacOS 11.7.7 (Big Sur). Both `~/.librewolf` and `~/.mozilla` folders don't exist.
After some trial and error, I was able to connect from LibreWolf+KeePassXC-Browser to my KeepassXC database copying the folder `~/Library/Application Support/Mozilla/NativeMessagingHosts` as a subfolder of `~/Library/Application Support/librewolf`.

**Reply by keizah7** [reply]
@nafarinha 

```bash
mkdir ~/Library/Application\ Support/librewolf/NativeMessagingHosts
```

KeepassXC/Settings/Browser Integration/Use custom../Config Location:
```
~/Library/Application Support/librewolf/NativeMessagingHosts
```

**Reply by j-lakeman** [reply]
I found that on a case-sensitive macOS, the folder has to be `~/Library/Application Support/LibreWolf/NativeMessagingHosts`.
I did not need to enable mentioned custom config location.

**Comment by magicdude4eva** [comment]
For Linux this will work -> https://support.1password.com/additional-browsers/?linux
